src/feature/relay/router.c

```diff
@@ -881,6 +881,43 @@ router_write_fingerprint(int hashed)
   return result;
 }
 
+STATIC int
+router_write_ed25519_identity(void)
```

Comment:
Rather than copying most of the code from router_write_fingerprint(), you could add an `int ed25519_identity` to router_write_fingerprint(), and re-use most of the code in that function?